app/lib/server/handler.ts

```diff
@@ -84,13 +85,18 @@ class Handler {
           }
         },
         onShardStored: (meta: CARMetadata) => {
-          this.#db.updateUser(this.#dbUser.id, {
-            ...this.#dbUser,
-            points: this.#dbUser.points + meta.size * mRachaPointsPerByte,
-          })
+          totalBytesUploaded += meta.size
         },
       })
 
+      // Only award points after successful backup completion
+      const pointsEarned = totalBytesUploaded * mRachaPointsPerByte
+      console.log(`total size uploaded: ${totalBytesUploaded} bytes`)
+      await this.#db.updateUser(this.#dbUser.id, {
+        ...this.#dbUser,
+        points: this.#dbUser.points + pointsEarned,
```

Comment:
Is this instance re-used? i.e. should we update `thgis.#dbUser.points` also?

Reply:
Should there be an increment points method? Is there a race condition here? Can the user have 2 backups running concurrently? i.e. at this point the `dbUser` is pretty old - could another backup have finished in the mean time?

If more than 1 backup can run concurrently for a user then we should at the least get the latest points value immediately before updating it to minimise chance of losing points.